examples/Components/Typeahead.purs

```diff
@@ -44,97 +38,90 @@ data Message
-            , available = RD.Success (filter (_ /= item) arr)
-            , search = ""
-            }
-          H.raise $ SelectionsChanged newSelections
-    S.Searched str -> do
-      st <- H.get
-      -- we'll use an external api to search locations
-      H.modify_ _ { available = RD.Loading }
-      items <- H.liftAff $ searchLocations str
-      H.modify_ _ { available = items <#> \xs -> difference xs st.selections }
-    _ -> pure unit
-
-  -- You can remove all type signatures except for this one; we need to tell the
-  -- compiler about the `a` type variable. The minimal necessary signature is below.
-  handleQuery :: forall a. Query a -> H.HalogenM _ _ _ _ _ (Maybe a)
-  handleQuery = case _ of
+  useLifecycleEffect do
+    void $ selectEvents.setCallback $ Just \_ val -> case val of
+      SelectedIndex ix -> do
+        available' <- Hooks.get availableId
+        for_ available' \arr ->
+          for_ (arr !! ix) \item -> do
+            selections' <- Hooks.get selectionsId
+            let newSelections = item : selections'
+            Hooks.modify_ availableId $ const (RD.Success (filter (_ /= item) arr))
+            Hooks.put selectionsId newSelections
+            select.clearSearch
+            Hooks.raise tokens.outputToken $ SelectionsChanged newSelections
+
+      NewSearch str -> do
+        selections' <- Hooks.get selectionsId
+        -- we'll use an external api to search locations
+        Hooks.put availableId RD.Loading
+        items <- liftAff $ searchLocations str
+        Hooks.put availableId $ items <#> \xs -> difference xs selections'
+
+      _ -> pure unit
+
+    pure Nothing
+
+  Hooks.useQuery tokens.queryToken case _ of
     GetSelections reply -> do
-       st <- H.get
-       pure $ Just $ reply st.selections
-
-  handleAction
-    :: Action
-    -> H.HalogenM (S.State State) (S.Action Action) ChildSlots Message Aff Unit
-  handleAction = case _ of
-    Remove item -> do
-      st <- H.get
-      let newSelections = filter (_ /= item) st.selections
-      H.modify_ _ { selections = newSelections }
-      H.raise $ ItemRemoved item
-    HandleDropdown msg -> case msg of
-      D.SelectionChanged oldSelection newSelection -> do
-        st <- H.get
-        let
-          mkLocation str = { name: "User Added: " <> str, population: "1" }
-          newSelections = case oldSelection, newSelection of
-            Nothing, Nothing ->
-              Nothing
-            Nothing, Just str ->
-              Just (mkLocation str : st.selections)
-            Just str, Nothing ->
-              Just (filter (_ /= mkLocation str) st.selections)
-            Just old, Just new ->
-              Just (mkLocation new : (filter (_ /= mkLocation old) st.selections))
-        for_ newSelections \selections ->
-          H.modify_ _ { selections = selections }
-
-  render :: S.State State -> H.ComponentHTML (S.Action Action) ChildSlots Aff
-  render st =
+       selections' <- Hooks.get selectionsId
```

Comment:
It's OK to use `get` here, but it's not necessary. This could also just be `pure $ Just $ reply selections`.

Reply:
Also done.